File: src/sdk-reference/go/1/realtime/subscribe/snippets/user-notifications.go
// Avoid program exit before receiving notification
exitPrgm := make(chan bool)

// Subscribe to notifications for documents containing a 'name' property
filters := json.RawMessage(`{ "exists": "name" }`)

// Start an async listener
listener := make(chan types.KuzzleNotification)
go func() {
  notification := <-listener

  if notification.Type == "user" {
    fmt.Printf("Volatile data: %s\n", notification.Volatile)
    // Volatile data: {"sdkVersion":"1.0.0","username":"nina vkote"}
    fmt.Printf("Currently %d users in the room\n", notification.Result.Count)
  }

  exitPrgm <- true
}()

options := types.NewRoomOptions()
options.SetUsers(types.USERS_ALL)

_, err := kuzzle.Realtime.Subscribe(
	"nyc-open-data",
	"yellow-taxi",
	filters,
	listener,
	options)

if err != nil {
  log.Fatal(err)
}

// instantiate a second kuzzle client because
// the same sdk instance does not receive their own notifications
f := websocket.NewWebSocket("kuzzle", nil)
fuzzle, _ := kuzzlepkg.NewKuzzle(f, nil)

connectErr = fuzzle.Connect()
if connectErr != nil {
  log.Fatal(connectErr)
  os.Exit(1)
}

// Set some volatile data
fuzzle.SetVolatile(json.RawMessage(`{ "username": "nina vkote" }`))

// Subscribe to the same room with the second client
fuzzle.Realtime.Subscribe(
	"nyc-open-data",
	"yellow-taxi",
	filters,
	make(chan types.KuzzleNotification),
	nil)

<-exitPrgm

File: src/sdk-reference/java/1/realtime/subscribe/snippets/user-notifications.java
String filters = "{ \"exists\": \"name\" }";
RoomOptions options = new RoomOptions();
options.setUsers("all");

NotificationListener listener = new NotificationListener() {
  public void onMessage(NotificationResult notification) {
    System.out.println(notification.getVolatiles());
    // "{ "username\": "nina vkote" }"
    System.out.println("Currently " + notification.getResult().getCount() + " users in the room");
  }
};

try {
  kuzzle.getRealtime().subscribe(
    "nyc-open-data",
    "yellow-taxi",
    filters,
    listener,
    options
  );

  // Instantiates a second kuzzle client: multiple subscriptions
  // made by the same user will not trigger "new user" notifications
  WebSocket ws2 = new WebSocket("kuzzle");
  Kuzzle kuzzle2 = new Kuzzle(ws2);
  kuzzle2.connect();

  // Set some volatile data
  RoomOptions options2 = new RoomOptions();
  options2.setVolatiles("{ \"username\": \"nina vkote\" }");

  // Subscribe to the same room with the second client
  kuzzle2.getRealtime().subscribe(
    "nyc-open-data",
    "yellow-taxi",
    filters,
    listener,
    options2
  );
} catch (KuzzleException e) {
  System.err.println(e.getMessage());
}
